sqlalchemy-stubs/sql/schema.pyi

```diff
@@ -111,6 +117,12 @@ class Column(SchemaItem, ColumnClause[_T]):
                  nullable: bool = ..., onupdate: Any = ..., primary_key: bool = ..., server_default: Any = ...,
                  server_onupdate: Union[FetchedValue, FunctionElement] = ..., quote: Optional[bool] = ..., unique: bool = ...,
                  system: bool = ..., comment: str = ...) -> None: ...
+    @overload
+    def __init__(self, name: str, *args: Any, autoincrement: Union[bool, str] = ...,
```

Comment:
This overload is too lenient. Three comments here:
* Do you really need the `*args: Any`? This will make mypy accept calls like `Column('test', 42)` that fail at runtime.
* It would be great to infer `Column[None]` for this overload in the plugin (currently mypy will either infer `Column[Any]` or will ask for annotation).
* It is worth adding a tests for this (and maybe also for `ForeignKey` while you are at it).